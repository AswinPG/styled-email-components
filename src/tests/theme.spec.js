import React from 'react';
import { renderToStaticMarkup } from 'react-dom/server';
import styled, { withTheme, ThemeProvider } from '../';

describe('theme', () => {
  it('works with `styled.component`', () => {
    const Link = styled.a`
      color: ${props => props.theme.color};
    `;
    expect(
      renderToStaticMarkup(
        <ThemeProvider theme={{ color: 'green' }}>
          <Link>Link</Link>
        </ThemeProvider>,
      ),
    ).toMatchSnapshot();
  });

  it('works with `styled(Component)`', () => {
    const Heading = props => <h1 {...props} />;
    const StyledHeading = styled(Heading)`
      font-size: ${props => props.theme.size};
    `;
    expect(
      renderToStaticMarkup(
        <ThemeProvider theme={{ size: '30px' }}>
          <StyledHeading>Hey</StyledHeading>
        </ThemeProvider>,
      ),
    ).toMatchSnapshot();
  });

  it('works with `withTheme`', () => {
    const Component = ({ theme, innerRef, ...rest }) => <div {...rest} />;
    const ThemedComponent = withTheme(Component);
    const WrapperComponent = props => <ThemedComponent {...props} />;
    const StyledComponent = styled(WrapperComponent)``;
    expect(
      renderToStaticMarkup(
        <ThemeProvider theme={{ attribute: 42 }}>
          <StyledComponent />
        </ThemeProvider>,
      ),
    ).toMatchSnapshot();
  });
});
