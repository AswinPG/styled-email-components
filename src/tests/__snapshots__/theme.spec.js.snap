// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`theme works with \`styled(Component)\` 1`] = `"<h1 style=\\"font-size:30px\\">Hey</h1>"`;

exports[`theme works with \`styled.component\` 1`] = `"<a style=\\"color:green\\">Link</a>"`;

exports[`theme works with \`withTheme\` 1`] = `"<div></div>"`;
